### PR TITLE
Add metrics-emitter add-on

### PR DESCRIPTION
**Pull request by bodymindarts**
This is currently deployed in the lab.

### CODE REVIEW

File: manifests/server.yml [hunk 1]
```diff
@@ -39,8 +39,8 @@ instance_groups:
             username: management
             password: (( vault meta.vault "/rabbitmq/admin/management:password" ))
           broker:
-            username: broker
-            password: (( vault meta.vault "/rabbitmq/admin/broker:password" ))
+            username: (( grab meta.broker.username ))
+            password: (( grab meta.broker.password ))
```
Comment (thomasmitchell) [inline manifests/server.yml]: This should not be changed. These properties refer to the RabbitMQ users that the broker uses, whereas those values in meta refer to the authentication used to login to the broker itself.

File: manifests/broker.yml [hunk 2]
```diff
@@ -65,8 +65,8 @@ instance_groups:
           management:
             username: management
           administrator:
-            username: broker
-            password: (( vault meta.vault "/rabbitmq/admin/broker:password" ))
+            username: (( grab meta.broker.username ))
+            password: (( grab meta.broker.password ))
```
Comment (thomasmitchell) [inline manifests/broker.yml]: This should not be changed. These properties refer to the RabbitMQ users that the broker uses, whereas those values in meta refer to the authentication used to login to the broker itself.

File: manifests/addons/metrics-emitter.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          tls:
+            cert: (( vault meta.cf.exodus_path ":loggregator_tls_agent_cert" ))
+            key: (( vault meta.cf.exodus_path ":loggregator_tls_agent_key" ))
+            ca_cert: (( vault meta.cf.exodus_path ":loggregator_ca" ))
+
+##### Add new releases here
+releases:
+- name: rabbitmq-metrics-emitter
+  version: "0.3.2"
+  url: git+https://github.com/starkandwayne/rabbitmq-metrics-emitter-release
```
Comment (thomasmitchell) [inline manifests/addons/metrics-emitter.yml]: Can BOSH make a release from just a repo URL? I wouldn't think that this would work unless the release was already uploaded to the director through other means.

File: manifests/addons/metrics-emitter.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          user: management
+          password: (( vault meta.vault "/rabbitmq/admin/management:password" ))
+        loggregator:
+          tls:
+            cert: (( vault meta.cf.exodus_path ":loggregator_tls_agent_cert" ))
+            key: (( vault meta.cf.exodus_path ":loggregator_tls_agent_key" ))
+            ca_cert: (( vault meta.cf.exodus_path ":loggregator_ca" ))
+
+##### Add new releases here
+releases:
```
Comment (thomasmitchell) [inline manifests/addons/metrics-emitter.yml:36]: can you please add these releases to the `releases.yml` file, and then inject them here? Having all the releases in one place makes it easier to track version upgrades later on.

File: manifests/addons/metrics-emitter.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          ca_cert: (( vault meta.cf.exodus_path ":loggregator_ca" ))
+          agent:
+            cert: (( vault meta.cf.exodus_path ":loggregator_tls_agent_cert" ))
+            key: (( vault meta.cf.exodus_path ":loggregator_tls_agent_key" ))
+  - name: rabbitmq-metrics-emitter
+    release: rabbitmq-metrics-emitter
+    properties:
+      rabbitmq_metrics_emitter:
+        cloud_foundry:
+          api:    (( concat "https://api." meta.cf.system_domain ))
```
Comment (thomasmitchell) [inline manifests/addons/metrics-emitter.yml]: If you'd like, you can use `(( grab meta.cf.api_url ))` for this, as that's already defined

File: manifests/addons/metrics-emitter.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          skip_ssl_validation: true
```
Comment (thomasmitchell) [inline manifests/addons/metrics-emitter.yml:23]: Can skipping SSL validation be parameterized? We may actually use a trusted CA cert for the CF in many cases.

File: manifests/addons/metrics-emitter.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          password:   (( vault meta.cf.exodus_path ":admin_password" ))
+        rmq_management:
+          endpoint: (( concat "http://" params.rmq_domain ":15672/api" ))
+          user: management
+          password: (( vault meta.vault "/rabbitmq/admin/management:password" ))
+        loggregator:
+          tls:
+            cert: (( vault meta.cf.exodus_path ":loggregator_tls_agent_cert" ))
+            key: (( vault meta.cf.exodus_path ":loggregator_tls_agent_key" ))
+            ca_cert: (( vault meta.cf.exodus_path ":loggregator_ca" ))
```
Comment (thomasmitchell) [inline manifests/addons/metrics-emitter.yml:34]: out of curiosity, why is there a loggregator agent on the box, as well as creds for the tool to talk to loggregator directly?

File: manifests/addons/broker-tls.yml [hunk 8]
```diff
@@ -53,7 +53,3 @@ instance_groups:
     properties:
       broker:
         host: (( concat params.broker_domain ":443" ))
-
-meta:
-  broker:
-    protocol: https
```
Comment (thomasmitchell) [inline manifests/addons/broker-tls.yml]: Why was this removed? This is used to tell the route_registrar what scheme to register with the gorouter

**Review by thomasmitchell — CHANGES_REQUESTED** [review]

File: manifests/addons/metrics-emitter.yml (hunk 3 above)
Comment (dennisjbell) [reply manifests/addons/metrics-emitter.yml]: Genesis doesn't use the git lfs release mechanism -- please make a standard release.

File: manifests/addons/metrics-emitter.yml (hunk 7 above)
Comment (bodymindarts) [reply manifests/addons/metrics-emitter.yml:34]: Talking to loggregator happens via localhost if I remember correctly
